tests/Fixtures/Migrations/CreateAuthorTable.php

```diff
@@ -4,23 +4,33 @@
 
 namespace Tests\Tempest\Fixtures\Migrations;
 
+use Tempest\Database\DatabaseDriver;
 use Tempest\Database\Migration;
 use Tempest\Database\Query;
+use Tempest\Database\QueryStatement;
 
 final readonly class CreateAuthorTable implements Migration
 {
+    public function __construct(
+        private DatabaseDriver $driver,
+    ) {
+    }
+
     public function getName(): string
     {
         return '0000-00-00_create_author_table';
     }
 
     public function up(): Query|null
     {
-        return new Query("CREATE TABLE Author (
-            `id` INTEGER PRIMARY KEY AUTOINCREMENT,
-            `name` TEXT NOT NULL,
-            `type` TEXT
-        )");
+        return $this->driver->createQueryStatement('Author')
```

Comment:
I wonder if we need the closure? 

```php
$this->driver
    ->createQueryStatement('Author')
    ->primary()
    ->statement('name TEXT NOT NULL')
    ->statement('type TEXT')
    ->toQuery();
```

Reply:
the idea behind the closure is so that I can format the query easily. Maybe I can remove the need for it, but would require some work.

Reply:
Let's keep it for now, I'll probably look into refactoring it after the PR is merged :)